### PR TITLE
Update docs to add new columns to `mz_source_statistics`

### PR DESCRIPTION
**Pull request by moulimukherjee**
Update docs to add new columns to `mz_source_statistics`

### Motivation
Follow up from https://github.com/MaterializeInc/materialize/pull/19044, adding documentation for the columns added in that PR

### Tips for reviewer
Please do suggest if different wording would make more sense here.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: doc/user/content/sql/system-catalog/mz_internal.md [hunk 1]
```diff
@@ -204,6 +204,8 @@ the system are restarted.
...
 | `bytes_received`       | [`bigint`]     | The number of bytes the worker has read from the external system. Bytes are counted in a source type-specific manner and may or may not include protocol overhead.                                                                                                                  |
+| `envelope_state_bytes` | [`bigint`]     | The number of bytes stored in the source envelope state. Bytes are counted in a source type-specific manner and may or may not include protocol overhead.                                                                                                                           |
+| `envelope_state_count` | [`bigint`]     | The number of records stored in the source envelope state.                                                                                                                                                                                                                          |
```
Comment (guswynn) [inline doc/user/content/sql/system-catalog/mz_internal.md:208]: I'd talk more about how they are _estimates_ for now!

Comment (moulimukherjee) [reply doc/user/content/sql/system-catalog/mz_internal.md:208]: Updating.

The count of records will be exact though, and the bytes will be estimates.

Comment (moulimukherjee) [reply doc/user/content/sql/system-catalog/mz_internal.md]: Done!

File: doc/user/content/sql/system-catalog/mz_internal.md [hunk 2]
```diff
@@ -204,6 +204,8 @@ the system are restarted.
...
 | `updates_committed`    | [`bigint`]     | The number of updates (insertions plus deletions) the worker has committed to the storage layer.                                                                                                                                                                                    |
 | `bytes_received`       | [`bigint`]     | The number of bytes the worker has read from the external system. Bytes are counted in a source type-specific manner and may or may not include protocol overhead.                                                                                                                  |
+| `envelope_state_bytes` | [`bigint`]     | The esimated number of bytes stored in the source envelope state. Note, these are currently rough estimates and may not reflect the exact value.                                                                                                                                    |
```
Comment (tr0njavolta) [inline doc/user/content/sql/system-catalog/mz_internal.md:207]:
```suggestion
| `envelope_state_bytes` | [`bigint`]     | The estimated number of bytes stored in the source envelope state. Note, these are currently rough estimates and may not reflect the exact value.                                                                                                                                    |
```

**Review by tr0njavolta — CHANGES_REQUESTED** [review]
One typo, other than that 👍

File: doc/user/content/sql/system-catalog/mz_internal.md (hunk 2 above)
Comment (morsapaes) [inline doc/user/content/sql/system-catalog/mz_internal.md:207]:
```suggestion
| `envelope_state_bytes` | [`bigint`]     | The number of bytes stored in the source envelope state.                                                                       |
```

Comment (moulimukherjee) [reply doc/user/content/sql/system-catalog/mz_internal.md:207]: Thanks for catching this 🙈

Comment (moulimukherjee) [reply doc/user/content/sql/system-catalog/mz_internal.md:207]: Sounds good 👍

Comment (moulimukherjee) [reply doc/user/content/sql/system-catalog/mz_internal.md]: Resolving since the word is removed in the next commit

**Pull request merged**